Src/ILGPU/ArrayView.cs

```diff
@@ -229,11 +229,7 @@ public ArrayView(ArrayViewSource source, long index, long length)
         {
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
             [ViewIntrinsic(ViewIntrinsicKind.GetViewElementAddressByIndex)]
-            get
-            {
-                IndexTypeExtensions.AssertIntIndex(Length);
```

Comment:
QUESTION: Don't we have to adapt these assertions to match the ones you have added to the code generator?

Reply:
The assertion already exists in the ArrayView's indexer that accepts a `long` parameter.

https://github.com/m4rs-mt/ILGPU/blob/20342179ec67ffddce8bf9d421339cc72f4bbac2/Src/ILGPU/ArrayView.cs#L264-L277